### PR TITLE
nexmo-dotnet-code-snippets Update

### PR DESCRIPTION
**Pull request by NexmoDev**

### CODE REVIEW

**Comment by mheap** [comment]
@tbedford Did you check all code snippets that have changed, or a sample?

**Comment by tbedford** [comment]
@mheap : only a sample. I can set some time aside today to go through them all though. I don't have a Windows/.NET setup as present though for testing the actual code.

**Comment by mheap** [comment]
Please go through all of them. We don't have to execute the code this time, but it's something we should bear in mind for the future as we'll be getting more and more .NET contributions

File: _examples/voice/make-an-outbound-call-with-ncco/csharp.yml [hunk 1]
```diff
@@ -8,9 +8,9 @@ client:
     from_line: 17
     to_line: 23
 code:
-    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Controllers/VoiceController.cs
-    from_line: 78
-    to_line: 111
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Senders/VoiceSender.cs
```
Comment (tbedford) [inline _examples/voice/make-an-outbound-call-with-ncco/csharp.yml]: Numbering appears off - start of function is missed which makes the code displayed look off.

File: _examples/application/update-application/csharp.yml [hunk 2]
```diff
@@ -4,12 +4,12 @@ language: dotnet
...
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Senders/ApplicationSender.cs
+    from_line: 58
+    to_line: 76
 client:
     source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Authentication/BasicAuth.cs
     from_line: 17
     to_line: 21
-file_name: ApplicationController.cs
+file_name: ApplicationSender.cs
 run_command: 'Run using your IDE'
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/application/update-application/csharp.yml]: fixed spelling error in code snippets

File: _examples/application/delete-application/csharp.yml [hunk 3]
```diff
@@ -4,12 +4,12 @@ language: dotnet
 dependencies:
     - Nexmo.Csharp.Client
 code:
-    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Controllers/ApplicationController.cs
-    from_line: 72
-    to_line: 74
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Senders/ApplicationSender.cs
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/application/delete-application/csharp.yml]: fixed spelling error in code snippets

File: _examples/verify/send-verification-request-with-workflow/csharp.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+title: .NET
+language: dotnet
+dependencies:
+    - Nexmo.Csharp.Client
+client:
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Authentication/BasicAuth.cs
+    from_line: 13
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/verify/send-verification-request-with-workflow/csharp.yml]: @tbedford can you double check that you have the latest version from master? pulling from master all the formatting in verify looks fine

File: _examples/verify/search-verification-request/csharp.yml [hunk 5]
```diff
@@ -8,9 +8,9 @@ client:
     from_line: 13
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/verify/search-verification-request/csharp.yml]: @tbedford can you double check that you have the latest version from master? pulling from master all the formatting in verify looks fine

File: _examples/verify/send-verification-request/csharp.yml [hunk 6]
```diff
@@ -8,9 +8,9 @@ client:
     from_line: 13
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/verify/send-verification-request/csharp.yml]: @tbedford can you double check that you have the latest version from master? pulling from master all the formatting in verify looks fine

File: _examples/verify/trigger-next-verification-process/csharp.yml [hunk 7]
```diff
@@ -8,9 +8,9 @@ client:
     from_line: 13
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/verify/trigger-next-verification-process/csharp.yml]: @tbedford can you double check that you have the latest version from master? pulling from master all the formatting in verify looks fine

File: _examples/voice/make-an-outbound-call-with-ncco/csharp.yml (hunk 1 above)
Comment (slorello89) [reply _examples/voice/make-an-outbound-call-with-ncco/csharp.yml]: @tbedford  - This is actually be design - it cuts off a different mechanism for generating the client, and the separate generate NCCO function is in place for organizational purposes. The formatting is actually consistent with what currently exists for the same snippet: https://developer.nexmo.com/voice/voice-api/code-snippets/make-an-outbound-call-with-ncco 
I'm currently working on refactoring NCCO generation in .NET which will allow a much more coherent mechanism for creating NCCOs so I will have to revist this snippet at that time anyway. Right now the only mechanism I have for making this coherent without those extra indents would be to fold the dynamic object creation in with everything else which would make that function a bit of a mess - so I'd like to keep it as is until I push out the NCCO updates if that's alright.

File: _examples/verify/send-verification-request-with-workflow/csharp.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tbedford) [reply _examples/verify/send-verification-request-with-workflow/csharp.yml]: I was using the review app.

File: _examples/voice/download-a-recording/csharp.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Authentication/FullAuth.cs
+    from_line: 17
+    to_line: 23
+code:
+    source: .repos/nexmo/nexmo-dotnet-code-snippets/NexmoDotnetCodeSnippets/Senders/VoiceSender.cs
+    from_line: 210
+    to_line: 210
+run_command: 'Run using your IDE'
+file_name: VoiceSender.cs
+unindent: true
```
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/voice/download-a-recording/csharp.yml]: Added a newline to the end of the file

File: _examples/verify/send-verification-request-with-workflow/csharp.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (slorello89) [reply _examples/verify/send-verification-request-with-workflow/csharp.yml]: This should be all set now - Apparently this has been out of whack in the Verify section for a while (I didn't actually modify any of the Client initialization in that pull request

**Comment by mheap** [comment]
@tbedford I've just destroyed the old app via the Heroku CLI and triggered a review app, so it should be live in 10 mins

**Pull request merged**